SharkCage/CageConfigurator/CageConfigurator.cs

```diff
@@ -1,19 +1,269 @@
+            public struct TRUSTEE : IDisposable
+            {
+                public IntPtr pMultipleTrustee;
+                public MULTIPLE_TRUSTEE_OPERATION MultipleTrusteeOperation;
+                public TRUSTEE_FORM TrusteeForm;
+                public TRUSTEE_TYPE TrusteeType;
+                public IntPtr ptstrName;
+
+                void IDisposable.Dispose()
+                {
+                    if (ptstrName != IntPtr.Zero) Marshal.Release(ptstrName);
+                }
+
+                public string Name { get { return Marshal.PtrToStringAuto(ptstrName); } }
+            }
+
+            [DllImport("advapi32.dll", SetLastError = true)]
+            public static extern bool AllocateAndInitializeSid(
+                ref SidIdentifierAuthority pIdentifierAuthority,
+                byte nSubAuthorityCount,
+                uint dwSubAuthority0, uint dwSubAuthority1,
+                uint dwSubAuthority2, uint dwSubAuthority3,
+                uint dwSubAuthority4, uint dwSubAuthority5,
+                uint dwSubAuthority6, uint dwSubAuthority7,
+                out IntPtr pSid);
+
+            [DllImport("advapi32.dll")]
+            public static extern IntPtr FreeSid(IntPtr pSid);
+
+            [DllImport("advapi32.dll", SetLastError = true)]
+            [return: MarshalAs(UnmanagedType.Bool)]
+            public static extern bool EqualSid(IntPtr pSid1, IntPtr pSid2);
+
+            [StructLayout(LayoutKind.Sequential)]
+            public struct SidIdentifierAuthority
+            {
+                [MarshalAs(UnmanagedType.ByValArray, SizeConst = 6, ArraySubType = UnmanagedType.I1)]
+                public byte[] Value;
+            }
+        }
+
         /// <summary>
         /// The main entry point for the application.
         /// </summary>
         [STAThread]
         static void Main(string[] parameter)
         {
-            Application.EnableVisualStyles();
-            Application.SetCompatibleTextRenderingDefault(false);
-            Application.Run(new CageConfiguratorForm(parameter.Length > 0 ? parameter[0] : null));
+            bool isReleaseMode = true;
+
+#if DEBUG
+            isReleaseMode = false;
+#endif
+
+            if (isReleaseMode && !StartedInCage())
+            {
+
+                MessageBox.Show("CageConfigurator can only be run on a secure desktop. Please start the Configurator manually with the corresponding entry in the CageChooser.",
+                    "SharkCage", MessageBoxButtons.OK, MessageBoxIcon.Information);
+            }
+            else
+            {
+                Application.EnableVisualStyles();
+                Application.SetCompatibleTextRenderingDefault(false);
+                Application.Run(new CageConfiguratorForm(parameter.Length > 0 ? parameter[0] : null));
+            }
+        }
+
+        // FIXME might be easier to ask the service for this info in the future? :/
```

Comment:
Maybe we should create an issue for that? Otherwise we can remove the FIXME.

Reply:
#96